ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeer.java

```diff
@@ -108,7 +108,10 @@ public EthPeer(
 
     final boolean supportsRequestId =
         getAgreedCapabilities().stream()
-            .anyMatch(capability -> capability.compareTo(EthProtocol.ETH66) >= 0);
+            .anyMatch(
+                capability ->
+                    EthProtocol.NAME.equals(capability.getName())
+                        && capability.compareTo(EthProtocol.ETH66) >= 0);
```

Comment:
```suggestion
                        && capability.getVersion() >= EthProtocol.ETH66.getVersion());
```

Reply:
I don't like this specific knowledge being in the EthPeers class. I've created a method in EthProtocol to do this. It will also make it easier to test this.